File: II - Linguaggi di programmazione e laboratorio/Java/Exercises/43_compito_java_20230221/Medico.java
public class Medico extends Operatore {
    private String specialita;
    private int n_visite_mese_precedente;
    private String ambulatorio;

    public Medico(int codice, String tipo, String nome_operatore, double costo_visita, String specialita,
            int n_visite_mese_precedente, String ambulatorio) {
        super(codice, tipo, nome_operatore, costo_visita);
        this.specialita = specialita;
        this.n_visite_mese_precedente = n_visite_mese_precedente;
        this.ambulatorio = ambulatorio;
    }

    public String toString() {
        return super.toString() + "\t" + specialita + "\t" + n_visite_mese_precedente + "\t" + ambulatorio
                + "\t-\t-\t-\tmedico";
    }
}

File: II - Linguaggi di programmazione e laboratorio/Esame/Python/43_compito_java_20230221/Medico.py
from Operatore import Operatore


class Medico(Operatore):
    def __init__(self, codice, tipo, nome, costo_visita, specialita, n_visite_mese_precedente, ambulatorio):
        super().__init__(codice, tipo, nome, costo_visita)
        self._specialita = specialita
        self._n_visite_mese_precedente = n_visite_mese_precedente
        self._ambulatorio = ambulatorio

    def __str__(self):
        return super().__str__() + "\t" + self._specialita + "\t" + str(self._n_visite_mese_precedente) + "\t" + self._ambulatorio + "\t-\t-\t-\tmedico"
